src/java/org/apache/cassandra/db/Directories.java

```diff
@@ -274,6 +292,77 @@ public boolean accept(File file)
         }
     }
 
+    public static void scheduleVerifyingDiskDoesNotExceedThresholdChecks()
+    {
+        ScheduledExecutors.scheduledTasks.scheduleAtFixedRate(
+                                getVerifyDiskHasEnoughUsableSpaceRunnable(), 1, 1, TimeUnit.MINUTES);
+    }
+
+    @VisibleForTesting
+    static Runnable getVerifyDiskHasEnoughUsableSpaceRunnable()
+    {
+        return () -> {
+            try {
+                Directories.verifyDiskHasEnoughUsableSpace();
+            } catch (ExceededDiskThresholdException e) {
+                // If node is already disabled propagating the exception from scheduled executor is redundant and clogs logs.
+                if (!StorageService.instance.isNodeDisabled()) {
+                    throw e;
+                }
+            }
+        };
+    }
+
+    public static void verifyDiskHasEnoughUsableSpace()
+    {
+        Map.Entry<DataDirectory, Double> maxPathToUtilization = getMaxPathToUtilization();
+        double threshold = DatabaseDescriptor.getMaxDiskUtilizationThreshold();
+        double currUtilization = maxPathToUtilization.getValue();
+        if (currUtilization >= threshold)
+        {
+            throw new ExceededDiskThresholdException(maxPathToUtilization.getKey().location, currUtilization, threshold);
+        }
+
+        if (StorageService.instance.isNodeDisabled() && StorageService.instance.isSetupCompleted())
```

Comment:
proposal: we should maybe add some padding here (like, maybe require that disk drops 2+% below threshold), otherwise we might get into a state where cassandra is constantly turning itself on and off right around the 99% mark.

Reply:
Do you think this margin should be configurable as well or just hardcoded at 2%?

Reply:
Hardcoded is probably fine.  Would do 2%